src/polyfillstub/intersection-observer-stub.js

```diff
@@ -60,9 +73,25 @@ export function upgradePolyfill(win, installer) {
   // instance deployed in v0.js vs the polyfill extension.
   const Stub = /** @type {typeof IntersectionObserverStub} */ (win.IntersectionObserver);
   if (Stub && UPGRADERS in Stub) {
+    const NativeIntersectionObserver =
+      win.IntersectionObserver && win.IntersectionObserver.native;
     delete win.IntersectionObserver;
     delete win.IntersectionObserverEntry;
     installer();
+    const PolyfilledIntersectionObserver = win.IntersectionObserver;
+    // NativeIntersectionObserver only exists in the case of an InOb
+    // that doesn't support {root: document}. In this case we'd like to
+    // select the fastest InOb we can on a per-instance basis.
+    if (NativeIntersectionObserver) {
+      win.IntersectionObserver = function (ioCallback, opts) {
+        if (opts && opts.root && opts.root.nodeType === 9) {
```

Comment:
Same:

> Should we also check if it's in an iframe? E.g. opts.root === window.document && !isIframed(window) is mostly equivalent to the opts.root == null.

We could even have a function here that can be used in both places. E.g. `simplifyOptions()` or `isOptionsRootNecessary()` or such.